lib/src/buttons/naver_auth_button.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2021 Talat El Beick. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:auth_buttons/src/helpers/auth_resolved_background_color.dart';
+import 'package:auth_buttons/src/helpers/auth_resolved_foreground_color.dart';
+import 'package:auth_buttons/src/helpers/auth_resolved_icon_url.dart';
+import 'package:auth_buttons/src/shared/auth_button_style.dart';
+import 'package:auth_buttons/src/shared/base/contracts/auth_button.dart';
+import 'package:auth_buttons/src/shared/core/widgets/auth_icon.dart';
+import 'package:auth_buttons/src/utils/auth_colors.dart';
+import 'package:auth_buttons/src/utils/auth_icons.dart';
+import 'package:auth_buttons/src/utils/auth_type.dart';
+import 'package:flutter/material.dart';
+
+/// create naver authentication button with multiple styles.
+///
+/// default style.
+///![](https://raw.githubusercontent.com/elbeicktalat/flutter_auth_buttons/master/doc/api/buttons/naver.png)
+///
+/// secondary style.
+///![](https://raw.githubusercontent.com/elbeicktalat/flutter_auth_buttons/master/doc/api/buttons/naver-secondary.png)
+///
+/// icon style.
+///![](https://raw.githubusercontent.com/elbeicktalat/flutter_auth_buttons/master/doc/api/buttons/naver-icon.png)
+class NaverAuthButton extends AuthButton {
+  const NaverAuthButton({
+    super.key = const ValueKey<String>('NaverAuthButton'),
+    super.onPressed,
+    super.onLongPress,
+    super.onHover,
+    super.onFocusChange,
+    super.focusNode,
+    super.autofocus,
+    super.text = 'Sign in with Naver',
+    @Deprecated(
+      'Use ThemeMode instead. '
+      'This property has no more effect. '
+      'This feature was deprecated after v3.0.0',
+    )
+    bool darkMode = false,
```

Comment:
Since it's new widget there is no need to add this deprecated code.

Reply:
Done!